django/applications/catmaid/migrations/0006_add_history_table_support.py

```diff
@@ -0,0 +1,501 @@
+        IF copy_inheritance THEN
+            -- If the table inherits from another table and <copy_inheritance> is
+            -- true, the complete inheritance hierarchy will be recreated for the
+            -- new table. Triggers, however, are only applied to the passed in live
+            -- table. Recursively walk parents to guarantee path to root.
+            RAISE NOTICE 'START INHERITANCE for %', live_table_name;
+            BEGIN
+                SELECT parent_schemaname, parent_tablename, parent_oid INTO STRICT parent_info
+                FROM catmaid_inheritening_tables
+                WHERE child_oid = live_table_name
+                AND child_schemaname = live_table_schema::text;
+                EXCEPTION
+                    WHEN NO_DATA_FOUND THEN
+                        -- Do nothing
+                    WHEN TOO_MANY_ROWS THEN
+                        -- Multi-inheritance support isn't implemented for histoty tables, yet
+                        RAISE EXCEPTION 'Couldn''t create history table, found more than one parent of %s.%s', live_table_schema, live_table_name;
+            END;
+
+            IF FOUND THEN
+                RAISE NOTICE 'Parent: %, %, %', parent_info.parent_schemaname, parent_info.parent_tablename, parent_info.parent_oid;
+                -- Recursively create a history table for the parent, without adding triggers
+                PERFORM create_history_table(parent_info.parent_schemaname, parent_info.parent_oid, TRUE, TRUE);
+            END IF;
+            RAISE NOTICE 'END INHERITANCE';
+        END IF;
+
+        IF parent_info IS NOT NULL THEN
+            parent_history_table_name = history_table_name(parent_info.parent_oid);
+            RAISE NOTICE 'CREATE History table with INHERITANCE %', parent_history_table_name;
+            -- Parent rows are sorted by their depth, most distant first.
+            -- If this parent table doesn't have a history table, yet, it is created.
+            -- Create a regular history table without inheritance, either
+            -- because no parent is available or no parent check was performed.
+            EXECUTE format(
+                'CREATE TABLE IF NOT EXISTS %I (LIKE %s) INHERITS (%I)',
+                history_table_name,live_table_name, parent_history_table_name
+            );
+        ELSE
+            -- Create a regular history table without inheritance, either
+            -- because no parent is available or no parent check was performed.
+            EXECUTE format(
+                'CREATE TABLE IF NOT EXISTS %I (LIKE %s)',
+                history_table_name,live_table_name
+            );
+        END IF;
+
+        -- Make all history columns (except the later added sys_period column
+        -- default to NULL
+        FOR column_info IN
+            SELECT c.column_name
+            FROM information_schema.columns c
+            WHERE c.table_schema NOT IN ('information_schema', 'pg_catalog')
```

Comment:
Yeah, I'm almost finished with the static readthrough and my only major overall suggestion was going to be "use regclass everywhere possible".